server/api/classyear/classyear.controller.js

```diff
@@ -220,5 +222,20 @@ function generateCode(codeLength){
       var character = (Math.floor(Math.random() * characterOptions.length));
       code = code.concat(characterOptions[character.toString()]);
   }
+  ClassYear.findOne({"dayCodes.code":code})
```

Comment:
this code should not be inside of the generateCode function. generate code is supposed to only create the code. We should do the checks outside of the function. Or at least not attempt to send error messages inside this function. you have response sending in the function but the response object isn't passed into generateCode. This will cause an exception if there ever is a duplicate code.